### PR TITLE
fix: bookmarks and completed

### PR DESCRIPTION
**Pull request by JuliusKryger**

### CODE REVIEW

File: src/CitMovie.Data/RepositoryInterfaces/IBookmarkRepository.cs [hunk 1]
```diff
@@ -1,40 +1,9 @@
-namespace CitMovie.Data;
```
Comment (sofushn) [inline src/CitMovie.Data/RepositoryInterfaces/IBookmarkRepository.cs:1]: I would guess this shouldn't have been removed

File: src/CitMovie.Api/Controller/CompletedController.cs [hunk 2]
```diff
@@ -75,47 +76,8 @@ public async Task<IActionResult> GetUserCompleted(int userId, int page = 0, int
...
-        if (completed == null)
-            return NotFound();
-
-        if (completed.UserId != userId)
-            return Forbid();
-
-        var deleted = await _completedManager.DeleteCompletedAsync(id);
-        return deleted ? NoContent() : NotFound();
-    }
-
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/CompletedController.cs:115]: Not really sure why this has been removed? should it not be possible to delete a movie if you added it by mistake or update it?

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 3]
```diff
@@ -34,8 +34,8 @@ public async Task<IActionResult> CreateBookmark(int userId, [FromBody] CreateBoo
             return BadRequest("Bookmark data is required.");
 
         createBookmarkDto.UserId = userId;
-        var createdBookmark = await _bookmarkManager.CreateBookmarkAsync(createBookmarkDto);
-        return CreatedAtAction(nameof(GetBookmark), new { userId, id = createdBookmark.BookmarkId }, createdBookmark);
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:38]: Why was this changed? The endpoint should return `created` status code not `ok`?

File: src/CitMovie.Api/Controller/BookmarkController.cs [hunk 4]
```diff
@@ -108,8 +108,8 @@ public async Task<IActionResult> DeleteBookmark(int userId, int id)
         if (bookmark.UserId != userId)
             return Forbid();
 
-        var deleted = await _bookmarkManager.DeleteBookmarkAsync(id);
-        return deleted ? NoContent() : NotFound();
```
Comment (sofushn) [inline src/CitMovie.Api/Controller/BookmarkController.cs:112]: Why was this changed?

File: src/CitMovie.Data/Repositories/CompletedRepository.cs [hunk 5]
```diff
@@ -7,13 +9,38 @@ public class CompletedRepository : ICompletedRepository
...
+    {
+        UserId = userId,
+        MediaId = mediaId,
+        Rewatchability = rewatchability,
+        Note = note,
+        User = user
+    };
+
+    return completed;
+}
```
Comment (sofushn) [inline src/CitMovie.Data/Repositories/CompletedRepository.cs:42]: Looks good, like that you use the SqlParameters

File: src/CitMovie.Data/Repositories/BookmarkRepository.cs [hunk 6]
```diff
@@ -29,42 +61,6 @@ await _context.Bookmarks
...
-        {
-            return false;
-        }
-    }
-
-     public async Task<int> GetTotalUserBookmarksCountAsync(int userId)
-    {
-        return await _context.Bookmarks.CountAsync(b => b.UserId == userId);
-    }
-}
```
Comment (sofushn) [inline src/CitMovie.Data/Repositories/BookmarkRepository.cs:70]: What's the reason for removing this?

**Review by sofushn — CHANGES_REQUESTED** [review]

File: src/CitMovie.Data/RepositoryInterfaces/IBookmarkRepository.cs (hunk 1 above)
Comment (JuliusKryger) [reply src/CitMovie.Data/RepositoryInterfaces/IBookmarkRepository.cs:1]: Hmmm ... Tested it everything works fine but no, that is not intentionel.

File: src/CitMovie.Data/Repositories/BookmarkRepository.cs (hunk 6 above)
Comment (JuliusKryger) [reply src/CitMovie.Data/Repositories/BookmarkRepository.cs:70]: I did not, I just changed them if you look above.

File: src/CitMovie.Api/Controller/CompletedController.cs (hunk 2 above)
Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/CompletedController.cs:115]: Will add it back.

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 4 above)
Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/BookmarkController.cs:112]: Beacuse if everything is succesfull i just wanna return an ok with the below string.

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 3 above)
Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/BookmarkController.cs:38]: CreatedAtAction was causing me some errors so i went for just returning a status code 200 with ok.
I can use return Created("Bookmark created successfully.", createBookmarkDto); which will return the 201 if that is important.

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 4 above)
Comment (sofushn) [reply src/CitMovie.Api/Controller/BookmarkController.cs:112]: But what if it wasn't, cant really see there has been added any logic to account for bookmark failing to update. Also the correct status code is `NoContent` not `Ok` since its an api and not a ui that needs to display a html page or something like that

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 3 above)
Comment (sofushn) [reply src/CitMovie.Api/Controller/BookmarkController.cs:38]: I think we should use the correct status code but `Created is fine if you cant get `CreatedAtAction` to work

File: src/CitMovie.Data/Repositories/BookmarkRepository.cs (hunk 6 above)
Comment (sofushn) [reply src/CitMovie.Data/Repositories/BookmarkRepository.cs:70]: Ahh my bad, though it was part of the move function you added 😅

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 3 above)
Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/BookmarkController.cs:38]: Created will still return 201 to my understanding, which is the same CreatedAtAction will. 

See this:
https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.controllerbase?view=aspnetcore-8.0

File: src/CitMovie.Api/Controller/BookmarkController.cs (hunk 4 above)
Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/BookmarkController.cs:112]: There is still logic for if it fails. 

```Csharp
if (userId != GetUserId())
            return Forbid();
            
var bookmark = await _bookmarkManager.GetBookmarkAsync(id);

if (bookmark == null) 
            return NotFound();
            
if (bookmark.UserId != userId)
            return Forbid();
```

Comment (JuliusKryger) [reply src/CitMovie.Api/Controller/BookmarkController.cs:112]: Changed it to respond with 204 instead of 200.

**Review by sofushn — APPROVED** [review]
